modules/Ecommerce/EcoDiscount/Repositories/EcoDiscountRepository.php

```
@@ -0,0 +1,255 @@
+        return $discount->load('products');
+    }
+
+    /**
+     * Attach products to a discount
+     */
+    public function attachProducts(string $discountId, array $productIds): void
+    {
+        $discount = $this->model->findOrFail($discountId);
+        $discount->products()->sync($productIds);
+    }
+
+    public function updateEcoDiscount(UuidInterface $id, array $data): bool
+    {
+        return $this->update($id, $data);
+    }
+
+    public function deleteEcoDiscount(UuidInterface $id): bool
+    {
+        return $this->delete($id);
+    }
+
+    public function getDiscountStatistics(): array
+    {
+        try {
+            $totalDiscounts = $this->model->count();
+            $activeDiscounts = $this->model->where('is_active', true)->count();
+            $expiredDiscounts = $this->model->where('end_date', '<', now())->count();
+            $usedDiscounts = $this->model->where('used_count', '>', 0)->count();
+
+            // Calculate total savings from orders
+            $totalSavings = DB::table('eco_orders')
+                ->whereNotNull('discount_id')
+                ->sum('discount_amount') ?? 0;
+
+            return [
+                'total_discounts' => [
+                    'number' => $totalDiscounts,
+                    'title' => 'إجمالي عدد التخفيضات',
+                ],
+                'active_discounts' => [
+                    'number' => $activeDiscounts,
+                    'title' => 'التخفيضات النشطة',
+                ],
+                'expired_discounts' => [
+                    'number' => $expiredDiscounts,
+                    'title' => 'التخفيضات المنتهية',
+                ],
+                'total_savings' => [
+                    'number' => number_format($totalSavings, 0),
+                    'title' => 'إجمالي المدخرات',
+                ]
+            ];
+        } catch (\Exception $e) {
+            // Fallback data
+            return [
```

Hardcoded fallback values should be avoided. Consider using configurable defaults or throwing the exception with proper error handling.